DataManagementSystem/scripts/dirac-dms-directory-sync.py

```diff
@@ -0,0 +1,552 @@
+    return S_ERROR('Directory deleting failed')
+  return S_OK('Removed directory successfully')
+
+def doUpload(fc, dm, result, source_dir, dest_dir, storage, delete, nthreads):
+  """
+  Wrapper for uploading files
+  """
+  if delete:
+    lfns = [dest_dir+"/"+filename for filename in result['Value']['Delete']['Files']]
+    if len(lfns)>0:
+      res = removeRemoteFiles(dm,lfns)
+      if not res['OK']:
+        gLogger.fatal('Deleting of files: ' + lfns + " -X- [FAILED]" +res['Message'])
+        DIRAC.exit( 1 )
+      else:
+        gLogger.notice("Deleting "+ ', '.join(lfns) + " -> [DONE]")
+
+    for directoryname in result['Value']['Delete']['Directories']:
+      res = removeRemoteDirectory(fc, dest_dir + "/" + directoryname)
+      if not res['OK']:
+        gLogger.fatal('Deleting of directory: ' + directoryname + " -X- [FAILED] " + res['Message'])
+        DIRAC.exit( 1 )
+      else:
+        gLogger.notice("Deleting "+ directoryname + " -> [DONE]")
+
+
+  for directoryname in result['Value']['Create']['Directories']:
+    res = createRemoteDirectory(fc, dest_dir+"/"+ directoryname)
+    if not res['OK']:
+      gLogger.fatal('Creation of directory: ' + directoryname + " -X- [FAILED] " + res['Message'])
+      DIRAC.exit( 1 )
+    else:
+      gLogger.notice("Creating " + directoryname + " -> [DONE]")
+
+  listOfFiles = result['Value']['Create']['Files']
+  #Chech that we do not have to many threads
+  if nthreads > len(listOfFiles):
+    nthreads = len(listOfFiles)
+
+  if nthreads == 0:
+    return S_OK('Upload finished successfully')
+
+  listOfListOfFiles = chunkList( listOfFiles, nthreads )
+  res = runInParallel(arguments=[dm, source_dir, dest_dir, storage], listOfLists=listOfListOfFiles, function=uploadListOfFiles )
+  if not res['OK']:
+    return S_ERROR("Upload of files failed")
+
+  return S_OK('Upload finished successfully')
+
+def uploadListOfFiles(dm, source_dir, dest_dir, storage, listOfFiles, tID):
+  """
+  Wrapper for multithreaded uploading of a list of files
+  """
+  log = gLogger.getSubLogger("[Thread %s] " % tID)
+  threadLine = "[Thread %s]" % tID
+  for filename in listOfFiles:
+    res = uploadLocalFile(dm, dest_dir+"/"+filename, source_dir+"/"+filename, storage)
+    if not res['OK']:
+      log.fatal(threadLine + ' Uploading ' + filename + ' -X- [FAILED] ' + res['Message'])
+      DIRAC.exit( 1 )
+    else:
+      log.notice(threadLine+ " Uploading " + filename + " -> [DONE]")
+
+def doDownload(dm, result, source_dir, dest_dir, delete, nthreads):
+  """
+  Wrapper for downloading files
+  """
+  if delete:
+    for filename in result['Value']['Delete']['Files']:
+      res = removeLocalFile(dest_dir+"/"+ filename)
+      if not res['OK']:
+        gLogger.fatal('Deleting of file: ' + filename + ' -X- [FAILED] ' + res['Message'])
+        DIRAC.exit( 1 )
+      else:
+        gLogger.notice("Deleting "+ filename + " -> [DONE]")
+
+    for directoryname in result['Value']['Delete']['Directories']:
```

Comment:
DIRAC coding conventions include Camel convention for naming variables:
directoryname -> directoryName